lesson18/src/js/main.js

```diff
@@ -0,0 +1,275 @@
+  slideArea.appendChild(pagenationArea).appendChild(fragment);
+}
+
+function addNoimage(){
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+  const sliderImage = createElementWithClassName("img", "slider__image");
+  sliderImage.src = './images/slide-image01.png'
+  sliderImage.width = 800;
+  sliderImage.height = 800;
+  sliderImage.style.opacity = 1;
+  slideImageArea.appendChild(sliderImage);
+}
+
+function switchPagination(currentIndex) {
+  const pagenationCurrent = document.getElementById("js-current");
+  pagenationCurrent.textContent = currentIndex;
+
+}
+
+function switchImg(target) {
+  const active = document.querySelector(".is-active");
+  active.classList.remove("is-active");
+  const changeSlide = document.querySelector(`[data-slide-index="${target}"]`);
+  changeSlide.classList.add("is-active");
+}
+
+function switcPagenationBullet(target){
+  const pagenationBullet = document.querySelector('[aria-current="true"');
+  pagenationBullet.setAttribute('aria-current', false);
+  const pageNations = document.querySelector(`[data-pagenation-index="${target}"]`);
+  pageNations.setAttribute("aria-current", true);
+
+}
+
+function clickPagenationBullet(){
+  const pagenationBullet = [...document.getElementsByClassName("bar")];
+  pagenationBullet.forEach((targetButton) => {
+    targetButton.addEventListener('click',(e) => {
+      const activePagenationBullet = document.querySelector('[aria-current="true"]');
+      const clickedPagenationBulletNo = e.currentTarget.getAttribute('data-pagenation-index');
+      activePagenationBullet.setAttribute('aria-current','false');
+      e.currentTarget.setAttribute('aria-current','true');
+      switchSlider(clickedPagenationBulletNo);
+      switchPagination(clickedPagenationBulletNo);
+      resetAutoPlaySlide();
+    })
+  })
+
+}
+
+function toggleButtonDisabled(index) {
+  const slideImages = [...document.getElementsByClassName("slider__item")];
+  const firstIndex = 1;
+  const lastIndex = slideImages.length;
+  const prevButton = document.getElementById("js-button_previous");
+  const nextButton = document.getElementById("js-button_next");
+
+  prevButton.disabled = Number(index) === firstIndex;
+  nextButton.disabled = Number(index) === lastIndex;
+}
+
+function renderSlidContents(slideImageData){
+  renderSlideImage(slideImageData);
+  renderPagenation(slideImageData.length);
+  renderPagenationBullet(slideImageData.length);
+  renderSlideButton();
+}
+
+function addSlideAction(){
+  addSwitchButtonEvent();
+  clickPagenationBullet();
+  autoPlayslide();
+}
+
+function callImageData() {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(
+        fetchImageData(
+//https://api.json-generator.com/templates/IZjWl012CAMD/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl : image = 0
+          //https://api.json-generator.com/templates/9tm12BO1y5Xx/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl&status=503 : 503
```

Comment:
Write the reason for not erasing

Reply:
I wrote the reason why I didn't delete the comment out.
I have a question,
Should I use a separate PR when issuing a confirmation API URL?

[850dce0](https://github.com/nyattsu72/hanson-frontend-cource/pull/26/commits/850dce06e9b82e21ff0dda258bea48aa284e307b)